Tasks/VsTestV2/inputparser.ts

```diff
@@ -528,8 +528,12 @@ function getTestPlatformPath(inputDataContract : idc.InputDataContract) {
 
     if (vsTestVersion.toLowerCase() === 'latest') {
         tl.debug('Searching for latest Visual Studio.');
-
-        let vstestconsolePath = getVSTestConsolePath('16.0', '17.0');
+        let vstestconsolePath = getVSTestConsolePath('17.0', '18.0');
```

Comment:
https://github.com/microsoft/azure-pipelines-tasks/blob/acc64cc7292c98597908325e53af9f898a896189/Tasks/VsTestV2/versionfinder.ts#L90 you'll need change here as well and test on the onprem flow.